File: app/src/main/java/ceui/lisa/activities/ViewPagerActivity.java
package ceui.lisa.activities;

import android.graphics.Color;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;
import android.view.View;


import com.ToxicBakery.viewpager.transforms.AccordionTransformer;
import com.ToxicBakery.viewpager.transforms.BackgroundToForegroundTransformer;
import com.ToxicBakery.viewpager.transforms.CubeInTransformer;
import com.ToxicBakery.viewpager.transforms.CubeOutTransformer;
import com.ToxicBakery.viewpager.transforms.DefaultTransformer;
import com.ToxicBakery.viewpager.transforms.DepthPageTransformer;
import com.ToxicBakery.viewpager.transforms.DrawerTransformer;
import com.ToxicBakery.viewpager.transforms.FlipHorizontalTransformer;
import com.ToxicBakery.viewpager.transforms.FlipVerticalTransformer;
import com.ToxicBakery.viewpager.transforms.ForegroundToBackgroundTransformer;
import com.ToxicBakery.viewpager.transforms.RotateUpTransformer;
import com.ToxicBakery.viewpager.transforms.ScaleInOutTransformer;
import com.ToxicBakery.viewpager.transforms.StackTransformer;
import com.ToxicBakery.viewpager.transforms.TabletTransformer;
import com.ToxicBakery.viewpager.transforms.ZoomInTransformer;
import com.ToxicBakery.viewpager.transforms.ZoomOutTransformer;

import java.util.ArrayList;
import java.util.List;

import ceui.lisa.R;
import ceui.lisa.fragments.FragmentSingleIllust;
import ceui.lisa.model.IllustsBean;
import ceui.lisa.utils.IllustChannel;

public class ViewPagerActivity extends BaseActivity {

    private List<IllustsBean> mIllusts = new ArrayList<>();

    @Override
    protected void initLayout() {
        getWindow().setStatusBarColor(Color.TRANSPARENT);
        getWindow().getDecorView().setSystemUiVisibility(
                View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN |
                        View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
        mLayoutID = R.layout.activity_view_pager;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void initView() {
        mIllusts.addAll(IllustChannel.get().getIllustList());
        ViewPager viewPager = findViewById(R.id.view_pager);
        viewPager.setPageTransformer(true, new DrawerTransformer());
        viewPager.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {
            @Override
            public Fragment getItem(int i) {
                return FragmentSingleIllust.newInstance(mIllusts.get(i));
            }

            @Override
            public int getCount() {
                return mIllusts.size();
            }
        });
        int position = getIntent().getIntExtra("position", 0);
        viewPager.setCurrentItem(position);
    }

    @Override
    protected void initData() {

    }
}

File: app/src/main/java/ceui/lisa/activities/ViewPagerActivity.kt
package ceui.lisa.activities

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.viewpager.widget.ViewPager
import ceui.lisa.R
import ceui.lisa.databinding.ActivityViewPagerBinding
import ceui.lisa.utils.DataChannel
import ceui.lisa.utils.PixivOperate.insertIllustViewHistory
import ceui.lisa.viewmodel.Dust
import com.ToxicBakery.viewpager.transforms.DrawerTransformer

class ViewPagerActivity : BaseActivity<ActivityViewPagerBinding>() {

    private lateinit var holder: Dust

    override fun hideStatusBar(): Boolean {
        return true
    }

    override fun initView() {
        holder = ViewModelProvider(this).get(Dust::class.java)
        holder.dust.value = ArrayList()
        holder.dust.value?.addAll(DataChannel.get().illustList)
        holder.index.observe(this, Observer{ index ->
            run {
                baseBind.viewPager.currentItem = index
            }
        })
        baseBind.viewPager.setPageTransformer(true, DrawerTransformer())
        baseBind.viewPager.addOnPageChangeListener(object :ViewPager.OnPageChangeListener{
            override fun onPageScrollStateChanged(state: Int) {
            }

            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
            }

            override fun onPageSelected(position: Int) {
                holder.index.value = position
                if (Shaft.sSettings.isSaveViewHistory) {
                    insertIllustViewHistory(holder.dust?.value!![position])
                }
            }
        })

        baseBind.viewPager.adapter = object : FragmentPagerAdapter(supportFragmentManager, 0) {
            override fun getItem(i: Int): Fragment {
//                return FragmentSingleIllust.newInstance(i)
                return Fragment()
            }

            override fun getCount(): Int {
                return holder.dust.value?.size!!
            }
        }
        val p = intent.getIntExtra("position", -1)
        if (p != -1) {
            holder.index.value = p
        }
    }

    override fun initLayout(): Int {
        return R.layout.activity_view_pager
    }

    override fun initData() {
    }
}
